src/components/views/settings/encryption/RecoveryPanelOutOfSync.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+import React, { JSX } from "react";
+import { Button } from "@vector-im/compound-web";
+import KeyIcon from "@vector-im/compound-design-tokens/assets/web/icons/key";
+
+import { SettingsSection } from "../shared/SettingsSection";
+import { _t } from "../../../../languageHandler";
+import { SettingsSubheader } from "../SettingsSubheader";
+import { accessSecretStorage } from "../../../../SecurityManager";
+
+interface RecoveryPanelOutOfSyncProps {
+    /**
+     * Callback for when the user has finished entering their recovery key.
+     */
+    onFinish: () => void;
+}
+
+/**
+ * This component is shown when the user secrets are not cached and the recovery key needs to be entered.
```

Comment:
```suggestion
 * This component is shown as part of the {@link EncryptionUserSettingsTab}, instead of the
 * {@link RecoveryPanel}, when some of the user secrets are not cached in the local client.
 *
 * It prompts the user to enter their recovery key so that the secrets can be loaded from 4S into
 * the client.
```